### PR TITLE
monitoring: bump TPO to v1.5.0

### PR DESCRIPTION
**Pull request by Quentin-M**
This bumps the TPO to `v1.5.0`, which includes the TPR->CRD migration for the `AppVersion` objects.

Note: this _needs to_ land in the next release, but should be low risk as the TPO's functionality didn't change in this release, just the TPR->CRD migration was implemented by bumping the vendored dependencies.

### CODE REVIEW

**Comment by Quentin-M** [comment]
This is imported from https://github.com/coreos/tectonic-installer/pull/1738 due to permission issues.

**Comment by Quentin-M** [comment]
image

image

```
=== RUN   Test
=== RUN   Test/Common
=== RUN   Test/Common/APIAvailable
=== RUN   Test/Cluster
=== RUN   Test/Cluster/AllNodesRunning
=== RUN   Test/Cluster/GetIdentityLogs
=== RUN   Test/Cluster/AllPodsRunning
=== RUN   Test/Cluster/KillAPIServer
=== RUN   Test/Cluster/AllResourcesCreated
--- PASS: Test (74.43s)
    --- PASS: Test/Common (0.44s)
        --- PASS: Test/Common/APIAvailable (0.44s)
        	common_test.go:20: API server is available.
    --- PASS: Test/Cluster (73.99s)
        --- PASS: Test/Cluster/AllNodesRunning (0.20s)
        	cluster_test.go:119: node ip-10-0-10-182.ca-central-1.compute.internal ready
        	cluster_test.go:119: node ip-10-0-27-69.ca-central-1.compute.internal ready
        	cluster_test.go:119: node ip-10-0-63-140.ca-central-1.compute.internal ready
        	cluster_test.go:119: node ip-10-0-8-242.ca-central-1.compute.internal ready
        	cluster_test.go:146: Successfully found 4 ready nodes.
        --- PASS: Test/Cluster/GetIdentityLogs (0.43s)
        	cluster_test.go:165: Successfully gathered identity logs.
        --- PASS: Test/Cluster/AllPodsRunning (0.31s)
        	cluster_test.go:86: All pods are ready.
        --- PASS: Test/Cluster/KillAPIServer (60.10s)
        	smoke_test.go:118: failed with error: old API server kube-apiserver-7xlct still running
        	smoke_test.go:119: retrying in 10s
        	smoke_test.go:118: failed with error: old API server kube-apiserver-7xlct still running
        	smoke_test.go:119: retrying in 10s
        	smoke_test.go:118: failed with error: old API server kube-apiserver-7xlct still running
        	smoke_test.go:119: retrying in 10s
        	smoke_test.go:118: failed with error: failed to get API server pod: Get https://qmachu-api.tectonic.dev.coreos.systems:443/api/v1/namespaces/kube-system/pods?labelSelector=k8s-app%3Dkube-apiserver: EOF
        	smoke_test.go:119: retrying in 10s
        	smoke_test.go:118: failed with error: API server has not yet been running for more than one check
        	smoke_test.go:119: retrying in 10s
        	cluster_test.go:269: API server pods successfully came back up.
        --- PASS: Test/Cluster/AllResourcesCreated (12.96s)
        	cluster_test.go:276: looking for resources defined by the provided manifests...
        	cluster_test.go:301: OK : [Kind: etcd.coreos.com/v1beta1:Cluster | Namespace: kube-system | Name: kube-etcd] - etcd/migrate-etcd-cluster.json
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: kube-system | Name: etcd-client-tls] - experimental/etcd-client-tls.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: kube-system | Name: etcd-operator] - experimental/etcd-operator.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: kube-system | Name: etcd-peer-tls] - experimental/etcd-peer-tls.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: kube-system | Name: etcd-server-tls] - experimental/etcd-server-tls.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: kube-system | Name: etcd-service] - experimental/etcd-service.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: kube-system | Name: kube-etcd-network-checkpointer] - experimental/kenc.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: kube-system | Name: kube-apiserver] - manifests/kube-apiserver-secret.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: kube-system | Name: kube-apiserver] - manifests/kube-apiserver.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: kube-system | Name: kube-cloud-cfg] - manifests/kube-cloud-config.yaml
        	cluster_test.go:301: OK : [Kind: policy/v1beta1:PodDisruptionBudget | Namespace: kube-system | Name: kube-controller-manager] - manifests/kube-controller-manager-disruption.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: kube-system | Name: kube-controller-manager] - manifests/kube-controller-manager-secret.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: kube-system | Name: kube-controller-manager] - manifests/kube-controller-manager.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: kube-system | Name: kube-dns] - manifests/kube-dns.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: kube-system | Name: kube-dns] - manifests/kube-dns.yaml
        	cluster_test.go:301: OK : [Kind: /v1:ConfigMap | Namespace: kube-system | Name: kube-flannel-cfg] - manifests/kube-flannel.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: kube-system | Name: kube-flannel] - manifests/kube-flannel.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: kube-system | Name: kube-proxy] - manifests/kube-proxy.yaml
        	cluster_test.go:301: OK : [Kind: policy/v1beta1:PodDisruptionBudget | Namespace: kube-system | Name: kube-scheduler] - manifests/kube-scheduler-disruption.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: kube-system | Name: kube-scheduler] - manifests/kube-scheduler.yaml
        	cluster_test.go:301: OK : [Kind: rbac.authorization.k8s.io/v1beta1:ClusterRoleBinding | Namespace:  | Name: system:default-sa] - manifests/kube-system-rbac-role-binding.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: kube-system | Name: pod-checkpointer] - manifests/pod-checkpointer.yaml
        	cluster_test.go:301: OK : [Kind: /v1:ConfigMap | Namespace: tectonic-system | Name: tectonic-config] - tectonic/config.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: tectonic-console] - tectonic/console/deployment.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: tectonic-console] - tectonic/console/service.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: etcd-operator] - tectonic/etcd/etcd-operator.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: kube-system | Name: heapster] - tectonic/heapster/deployment.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: kube-system | Name: heapster] - tectonic/heapster/service.yaml
        	cluster_test.go:301: OK : [Kind: /v1:ConfigMap | Namespace: tectonic-system | Name: tectonic-identity] - tectonic/identity/configmap.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: tectonic-identity] - tectonic/identity/deployment.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: tectonic-identity] - tectonic/identity/services.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: tectonic-identity-api] - tectonic/identity/services.yaml
        	cluster_test.go:301: OK : [Kind: /v1:ConfigMap | Namespace: tectonic-system | Name: tectonic-custom-error] - tectonic/ingress/default-backend/configmap.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: default-http-backend] - tectonic/ingress/default-backend/deployment.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: default-http-backend] - tectonic/ingress/default-backend/service.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: tectonic-lb] - tectonic/ingress/hostport/service.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:Ingress | Namespace: tectonic-system | Name: tectonic-ingress] - tectonic/ingress/ingress.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: tectonic-ingress-controller] - tectonic/ingress/nodeport/deployment.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: tectonic-lb] - tectonic/ingress/nodeport/service.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: tectonic-system | Name: alertmanager-main] - tectonic/monitoring/alertmanager-config.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: alertmanager-main] - tectonic/monitoring/alertmanager-service.yaml
        	cluster_test.go:301: OK : [Kind: monitoring.coreos.com/v1alpha1:Alertmanager | Namespace: tectonic-system | Name: main] - tectonic/monitoring/alertmanager.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: kube-system | Name: kube-controller-manager-prometheus-discovery] - tectonic/monitoring/kube-controller-manager-svc.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: kube-system | Name: kube-scheduler-prometheus-discovery] - tectonic/monitoring/kube-scheduler-svc.yaml
        	cluster_test.go:301: OK : [Kind: rbac.authorization.k8s.io/v1beta1:ClusterRoleBinding | Namespace:  | Name: kube-state-metrics] - tectonic/monitoring/kube-state-metrics-cluster-role-binding.yaml
        	cluster_test.go:301: OK : [Kind: rbac.authorization.k8s.io/v1beta1:ClusterRole | Namespace:  | Name: kube-state-metrics] - tectonic/monitoring/kube-state-metrics-cluster-role.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: kube-state-metrics] - tectonic/monitoring/kube-state-metrics-deployment.yaml
        	cluster_test.go:301: OK : [Kind: /v1:ServiceAccount | Namespace: tectonic-system | Name: kube-state-metrics] - tectonic/monitoring/kube-state-metrics-service-account.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: kube-state-metrics] - tectonic/monitoring/kube-state-metrics-service.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: node-exporter] - tectonic/monitoring/node-exporter-ds.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: node-exporter] - tectonic/monitoring/node-exporter-svc.yaml
        	cluster_test.go:301: OK : [Kind: rbac.authorization.k8s.io/v1beta1:ClusterRoleBinding | Namespace:  | Name: prometheus-k8s] - tectonic/monitoring/prometheus-k8s-cluster-role-binding.yaml
        	cluster_test.go:301: OK : [Kind: rbac.authorization.k8s.io/v1beta1:ClusterRole | Namespace:  | Name: prometheus-k8s] - tectonic/monitoring/prometheus-k8s-cluster-role.yaml
        	cluster_test.go:301: OK : [Kind: /v1:ConfigMap | Namespace: tectonic-system | Name: prometheus-k8s-rules] - tectonic/monitoring/prometheus-k8s-rules.yaml
        	cluster_test.go:301: OK : [Kind: /v1:ServiceAccount | Namespace: tectonic-system | Name: prometheus-k8s] - tectonic/monitoring/prometheus-k8s-service-account.yaml
        	cluster_test.go:301: OK : [Kind: monitoring.coreos.com/v1alpha1:ServiceMonitor | Namespace: tectonic-system | Name: alertmanager] - tectonic/monitoring/prometheus-k8s-service-monitor-alertmanager.yaml
        	cluster_test.go:301: OK : [Kind: monitoring.coreos.com/v1alpha1:ServiceMonitor | Namespace: tectonic-system | Name: kube-apiserver] - tectonic/monitoring/prometheus-k8s-service-monitor-apiserver.yaml
        	cluster_test.go:301: OK : [Kind: monitoring.coreos.com/v1alpha1:ServiceMonitor | Namespace: tectonic-system | Name: kube-controller-manager] - tectonic/monitoring/prometheus-k8s-service-monitor-kube-controller-manager.yaml
        	cluster_test.go:301: OK : [Kind: monitoring.coreos.com/v1alpha1:ServiceMonitor | Namespace: tectonic-system | Name: kube-scheduler] - tectonic/monitoring/prometheus-k8s-service-monitor-kube-scheduler.yaml
        	cluster_test.go:301: OK : [Kind: monitoring.coreos.com/v1alpha1:ServiceMonitor | Namespace: tectonic-system | Name: kube-state-metrics] - tectonic/monitoring/prometheus-k8s-service-monitor-kube-state-metrics.yaml
        	cluster_test.go:301: OK : [Kind: monitoring.coreos.com/v1alpha1:ServiceMonitor | Namespace: tectonic-system | Name: kubelet] - tectonic/monitoring/prometheus-k8s-service-monitor-kubelet.yaml
        	cluster_test.go:301: OK : [Kind: monitoring.coreos.com/v1alpha1:ServiceMonitor | Namespace: tectonic-system | Name: node-exporter] - tectonic/monitoring/prometheus-k8s-service-monitor-node-exporter.yaml
        	cluster_test.go:301: OK : [Kind: monitoring.coreos.com/v1alpha1:ServiceMonitor | Namespace: tectonic-system | Name: prometheus-operator] - tectonic/monitoring/prometheus-k8s-service-monitor-prometheus-operator.yaml
        	cluster_test.go:301: OK : [Kind: monitoring.coreos.com/v1alpha1:ServiceMonitor | Namespace: tectonic-system | Name: prometheus] - tectonic/monitoring/prometheus-k8s-service-monitor-prometheus.yaml
        	cluster_test.go:301: OK : [Kind: monitoring.coreos.com/v1alpha1:Prometheus | Namespace: tectonic-system | Name: k8s] - tectonic/monitoring/prometheus-k8s.yaml
        	cluster_test.go:301: OK : [Kind: rbac.authorization.k8s.io/v1beta1:ClusterRoleBinding | Namespace:  | Name: prometheus-operator] - tectonic/monitoring/prometheus-operator-cluster-role-binding.yaml
        	cluster_test.go:301: OK : [Kind: rbac.authorization.k8s.io/v1beta1:ClusterRole | Namespace:  | Name: prometheus-operator] - tectonic/monitoring/prometheus-operator-cluster-role.yaml
        	cluster_test.go:301: OK : [Kind: /v1:ServiceAccount | Namespace: tectonic-system | Name: prometheus-operator] - tectonic/monitoring/prometheus-operator-service-account.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: prometheus-operator] - tectonic/monitoring/prometheus-operator-svc.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: prometheus-operator] - tectonic/monitoring/prometheus-operator.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: prometheus] - tectonic/monitoring/prometheus-svc.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: tectonic-monitoring-auth-alertmanager] - tectonic/monitoring/tectonic-monitoring-auth-alertmanager-deployment.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: tectonic-monitoring-auth-alertmanager] - tectonic/monitoring/tectonic-monitoring-auth-alertmanager-svc.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: tectonic-monitoring-auth-prometheus] - tectonic/monitoring/tectonic-monitoring-auth-prometheus-deployment.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Service | Namespace: tectonic-system | Name: tectonic-monitoring-auth-prometheus] - tectonic/monitoring/tectonic-monitoring-auth-prometheus-svc.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: tectonic-system | Name: tectonic-monitoring-auth] - tectonic/monitoring/tectonic-monitoring-auth-secret.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:Ingress | Namespace: tectonic-system | Name: tectonic-monitoring-ingress] - tectonic/monitoring/tectonic-monitoring-ingress.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Namespace | Namespace:  | Name: tectonic-system] - tectonic/namespace.yaml
        	cluster_test.go:301: OK : [Kind: rbac.authorization.k8s.io/v1beta1:ClusterRoleBinding | Namespace:  | Name: admin-user] - tectonic/rbac/binding-admin.yaml
        	cluster_test.go:301: OK : [Kind: rbac.authorization.k8s.io/v1beta1:ClusterRoleBinding | Namespace:  | Name: discovery] - tectonic/rbac/binding-discovery.yaml
        	cluster_test.go:301: OK : [Kind: rbac.authorization.k8s.io/v1beta1:ClusterRole | Namespace:  | Name: admin] - tectonic/rbac/role-admin.yaml
        	cluster_test.go:301: OK : [Kind: rbac.authorization.k8s.io/v1beta1:ClusterRole | Namespace:  | Name: user] - tectonic/rbac/role-user.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: tectonic-system | Name: tectonic-ca-cert-secret] - tectonic/secrets/ca-cert.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: tectonic-system | Name: tectonic-identity-grpc-client-secret] - tectonic/secrets/identity-grpc-client.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: tectonic-system | Name: tectonic-identity-grpc-server-secret] - tectonic/secrets/identity-grpc-server.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: tectonic-system | Name: tectonic-ingress-tls-secret] - tectonic/secrets/ingress-tls.yaml
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: tectonic-system | Name: tectonic-license-secret] - tectonic/secrets/license.json
        	cluster_test.go:301: OK : [Kind: /v1:Secret | Namespace: tectonic-system | Name: coreos-pull-secret] - tectonic/secrets/pull.json
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: tectonic-stats-emitter] - tectonic/stats-emitter.yaml
        	cluster_test.go:301: OK : [Kind: apiextensions.k8s.io/v1beta1:CustomResourceDefinition | Namespace:  | Name: appversions.tco.coreos.com] - tectonic/updater/app-version-kind.yaml
        	cluster_test.go:301: OK : [Kind: tco.coreos.com/v1:AppVersion | Namespace: tectonic-system | Name: kubernetes] - tectonic/updater/app_versions/app-version-kubernetes.yaml
        	cluster_test.go:301: OK : [Kind: tco.coreos.com/v1:AppVersion | Namespace: tectonic-system | Name: tectonic-cluo-operator] - tectonic/updater/app_versions/app-version-tectonic-cluo.yaml
        	cluster_test.go:301: OK : [Kind: tco.coreos.com/v1:AppVersion | Namespace: tectonic-system | Name: tectonic-cluster] - tectonic/updater/app_versions/app-version-tectonic-cluster.yaml
        	cluster_test.go:301: OK : [Kind: tco.coreos.com/v1:AppVersion | Namespace: tectonic-system | Name: tectonic-monitoring] - tectonic/updater/app_versions/app-version-tectonic-monitoring.yaml
        	cluster_test.go:301: OK : [Kind: apiextensions.k8s.io/v1beta1:CustomResourceDefinition | Namespace:  | Name: migrationstatuses.kvo.coreos.com] - tectonic/updater/migration-status-kind.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: node-agent] - tectonic/updater/node-agent.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: kube-version-operator] - tectonic/updater/operators/kube-version-operator.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: tectonic-channel-operator] - tectonic/updater/operators/tectonic-channel-operator.yaml
        	cluster_test.go:301: OK : [Kind: apps/v1beta1:Deployment | Namespace: tectonic-system | Name: tectonic-cluo-operator] - tectonic/updater/operators/tectonic-cluo-operator.yaml
        	cluster_test.go:301: OK : [Kind: extensions/v1beta1:DeploymentOrDaemonSet | Namespace: tectonic-system | Name: tectonic-prometheus-operator] - tectonic/updater/operators/tectonic-prometheus-operator.yaml
        	cluster_test.go:301: OK : [Kind: tco.coreos.com/v1:ChannelOperatorConfig | Namespace: tectonic-system | Name: default] - tectonic/updater/tectonic-channel-operator-config.yaml
        	cluster_test.go:301: OK : [Kind: apiextensions.k8s.io/v1beta1:CustomResourceDefinition | Namespace:  | Name: channeloperatorconfigs.tco.coreos.com] - tectonic/updater/tectonic-channel-operator-kind.yaml
        	cluster_test.go:301: OK : [Kind: /v1:ConfigMap | Namespace: tectonic-system | Name: tectonic-monitoring] - tectonic/updater/tectonic-monitoring-config.yaml
        	cluster_test.go:337: all resources defined by the provided manifests are present
PASS
```

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.1 1.7.1-tectonic.1

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.1 1.7.1-tectonic.1

**Comment by Quentin-M** [comment]
Testing upgrade with https://github.com/coreos/tectonic-installer/pull/1742 combined.

**Comment by Quentin-M** [comment]
upgrade from 1.7.1-tectonic.1 1.7.1-tectonic.1

**Comment by Quentin-M** [comment]
TPO seems stuck:

```
Sync "tectonic-system/tectonic-monitoring" failed: waiting for Prometheus object changes failed: retrieving Prometheus status failed: retrieving stateful set failed: the server could not find the requested resource
```

**Comment by Quentin-M** [comment]
TPO seems stuck:

```
Sync "tectonic-system/tectonic-monitoring" failed: waiting for Prometheus object changes failed: retrieving Prometheus status failed: retrieving stateful set failed: the server could not find the requested resource
```

/cc @brancz

**Comment by Quentin-M** [comment]
upgrade from 1.7.1-tectonic.1 1.7.1-tectonic.1

**Comment by Quentin-M** [comment]
upgrade from 1.7.1-tectonic.1 1.7.1-tectonic.1

**Comment by Quentin-M** [comment]
upgrade from 1.7.1-tectonic.1 1.7.1-tectonic.1

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.1 1.7.1-tectonic.1

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.1 1.7.1-tectonic.1

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.1 1.7.1-tectonic.1

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.2 1.7.1-tectonic.2

**Comment by brancz** [comment]
There was indeed a problem with the TPO using the wrong API version for a request. We're in the process of resolving this (https://github.com/coreos-inc/tectonic-prometheus-operator/pull/111), but are facing some difficulties with the CI infra.

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.2 1.7.1-tectonic.2

**Comment by Quentin-M** [comment]
retest this please

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.2 1.7.1-tectonic.2

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.2 1.7.1-tectonic.2

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.2 1.7.1-tectonic.2

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.2 1.7.1-tectonic.2

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.2 1.7.1-tectonic.2

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.2 1.7.1-tectonic.2

**Comment by Quentin-M** [comment]
Got one from 1.7.1-tectonic.2:

```
NAME                                           STATUS    AGE       VERSION
ip-10-0-15-56.ca-central-1.compute.internal    Ready     18m       v1.7.3+coreos.0
ip-10-0-16-13.ca-central-1.compute.internal    Ready     19m       v1.7.3+coreos.0
ip-10-0-29-183.ca-central-1.compute.internal   Ready     18m       v1.7.3+coreos.0
ip-10-0-43-149.ca-central-1.compute.internal   Ready     18m       v1.7.3+coreos.0
```

```
kubectl get appversions.tco.coreos.com -n tectonic-system -o yaml
apiVersion: v1
items:
- apiVersion: tco.coreos.com/v1
  kind: AppVersion
  metadata:
    clusterName: ""
    creationTimestamp: 2017-08-22T19:39:19Z
    generation: 0
    name: kubernetes
    namespace: tectonic-system
    resourceVersion: "5319"
    selfLink: /apis/tco.coreos.com/v1/namespaces/tectonic-system/appversions/kubernetes
    uid: 96def58d-8771-11e7-a05b-06991b66f3f1
  spec:
    desiredVersion: 1.7.3+tectonic.1
    paused: false
  status:
    currentVersion: 1.7.3+tectonic.1
    paused: false
    targetVersion: ""
    taskStatuses:
    - name: Update kube-apiserver
      reason: ""
      state: Completed
    - name: Update kube-scheduler
      reason: ""
      state: Completed
    - name: Update kube-controller-manager
      reason: ""
      state: Completed
    - name: Update kube-proxy
      reason: ""
      state: Completed
    - name: Update Node Updater
      reason: ""
      state: Completed
    - name: Update tectonic-console
      reason: ""
      state: Completed
    - name: Update tectonic-identity
      reason: ""
      state: Completed
    - name: Update default-http-backend
      reason: ""
      state: Completed
    - name: Update tectonic-ingress-controller
      reason: ""
      state: Completed
    - name: prewarm-container-image-cache
      reason: ""
      state: Completed
- apiVersion: tco.coreos.com/v1
  kind: AppVersion
  metadata:
    annotations:
      tectonic-operators.coreos.com/upgrade-behaviour: CreateOrUpgrade
    clusterName: ""
    creationTimestamp: 2017-08-22T19:46:48Z
    generation: 0
    labels:
      managed-by-channel-operator: "true"
    name: tectonic-cluo-operator
    namespace: tectonic-system
    resourceVersion: "5504"
    selfLink: /apis/tco.coreos.com/v1/namespaces/tectonic-system/appversions/tectonic-cluo-operator
    uid: a27b768a-8772-11e7-a0bf-06991b66f3f1
  spec:
    desiredVersion: 0.1.3
    paused: false
  status:
    currentVersion: 0.1.3
    paused: false
    targetVersion: ""
    taskStatuses:
    - name: DaemonSet/tectonic-system/container-linux-update-agent-ds
      reason: ""
      state: Completed
    - name: Deployment/tectonic-system/container-linux-update-operator
      reason: ""
      state: Completed
- apiVersion: tco.coreos.com/v1
  kind: AppVersion
  metadata:
    clusterName: ""
    creationTimestamp: 2017-08-22T19:39:19Z
    generation: 0
    name: tectonic-cluster
    namespace: tectonic-system
    resourceVersion: "5738"
    selfLink: /apis/tco.coreos.com/v1/namespaces/tectonic-system/appversions/tectonic-cluster
    uid: 96e39cd7-8771-11e7-a05b-06991b66f3f1
  spec:
    desiredVersion: 1.7.3-tectonic.1
    paused: false
  status:
    currentVersion: 1.7.3-tectonic.1
    paused: false
    taskStatuses:
    - name: Update deployment tectonic-channel-operator
      reason: ""
      state: Completed
    - name: Update deployment kube-version-operator
      reason: ""
      state: Completed
    - name: Update deployment tectonic-cluo-operator
      reason: ""
      state: Completed
    - name: Update deployment tectonic-etcd-operator
      reason: ""
      state: Completed
    - name: Update deployment tectonic-prometheus-operator
      reason: ""
      state: Completed
    - name: Update AppVersion kubernetes
      reason: ""
      state: Completed
    - name: Update AppVersion tectonic-cluo-operator
      reason: ""
      state: Completed
    - name: Update AppVersion tectonic-etcd
      reason: ""
      state: Completed
    - name: Update AppVersion tectonic-monitoring
      reason: ""
      state: Completed
    - name: Cleanup unwanted old resources
      reason: ""
      state: Completed
- apiVersion: tco.coreos.com/v1
  kind: AppVersion
  metadata:
    clusterName: ""
    creationTimestamp: 2017-08-22T19:39:19Z
    deletionGracePeriodSeconds: null
    deletionTimestamp: null
    generation: 0
    name: tectonic-etcd
    namespace: tectonic-system
    resourceVersion: "5548"
    selfLink: /apis/tco.coreos.com/v1/namespaces/tectonic-system/appversions/tectonic-etcd
    uid: 96f0aaeb-8771-11e7-a05b-06991b66f3f1
  spec:
    desiredVersion: 0.0.1
    paused: false
  status:
    currentVersion: 0.0.1
    paused: false
- apiVersion: tco.coreos.com/v1
  kind: AppVersion
  metadata:
    clusterName: ""
    creationTimestamp: 2017-08-22T19:39:19Z
    generation: 0
    name: tectonic-monitoring
    namespace: tectonic-system
    resourceVersion: "5723"
    selfLink: /apis/tco.coreos.com/v1/namespaces/tectonic-system/appversions/tectonic-monitoring
    uid: 96f4e0a4-8771-11e7-a05b-06991b66f3f1
  spec:
    desiredVersion: 1.5.1
    paused: false
  status:
    currentVersion: 1.5.1
    paused: false
    targetVersion: ""
    taskStatuses:
    - name: Updating Prometheus Operator
      reason: ""
      state: Completed
    - name: Updating Prometheus
      reason: ""
      state: Completed
    - name: Updating Alertmanager
      reason: ""
      state: Completed
    - name: Updating node-exporter
      reason: ""
      state: Completed
    - name: Updating kube-state-metrics
      reason: ""
      state: Completed
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

**Comment by Quentin-M** [comment]
test upgrade from 1.7.1-tectonic.2 1.7.1-tectonic.2

**Comment by Quentin-M** [comment]
Got two other good upgrades from 1.7.1-tectonic.2 using a personal cluster (three masters, one worker) and the automated test.

File: config.tf [hunk 1]
```diff
@@ -57,10 +57,10 @@ variable "tectonic_container_images" {
     prometheus_operator          = "quay.io/coreos/prometheus-operator:v0.11.0"
     stats_emitter                = "quay.io/coreos/tectonic-stats:6e882361357fe4b773adbf279cddf48cb50164c1"
     stats_extender               = "quay.io/coreos/tectonic-stats-extender:487b3da4e175da96dabfb44fba65cdb8b823db2e"
-    tectonic_channel_operator    = "quay.io/coreos/tectonic-channel-operator:0.4.3"
+    tectonic_channel_operator    = "quay.io/coreos/tectonic-channel-operator:0.5.0"
     tectonic_etcd_operator       = "quay.io/coreos/tectonic-etcd-operator:v0.0.1"
     tectonic_monitoring_auth     = "quay.io/coreos/tectonic-monitoring-auth:v0.0.1"
-    tectonic_prometheus_operator = "quay.io/coreos/tectonic-prometheus-operator:v1.4.1"
+    tectonic_prometheus_operator = "quay.io/coreos/tectonic-prometheus-operator-dev:477fa943"
```
Comment (yifan-gu) [inline config.tf]: Didn't we have a new release?

Comment (Quentin-M) [inline config.tf]: This is the release that fixes the upgrade bug. @brancz noted that they will have another bug fix coming this week, which will likely result in 1.5.1 / 1.5.2 actually being cut officially and we will replace that image in a later RC.

**Comment by brancz** [comment]
We fixed the other bug found by QA, and cut the v1.5.1 release of the TPO.

**Pull request merged**